File: haskell/src/Solutions/Euler073.hs
module Solutions.Euler073 where

-- ceiling (a / b) = floor ((a + b - 1) / b)
-- -2, since this counts 1/2 and 1/3
count = length [() |
    den <- [2 .. 12000],
    num <- [(den + 2) `div` 3 .. den `div` 2],
    gcd den num == 1] - 2

euler073 :: IO String
euler073 = return $ show count

File: rust/src/solutions/euler073.rs
pub fn euler073() -> String {
    let mut ans = -2; // -2, since this counts 1/2 and 1/3
    for den in 2..=12000 {
        for num in (den + 2) / 3..=den / 2 {
            if num::integer::gcd(den, num) == 1 {
                ans += 1;
            }
        }
    }
    ans.to_string()
}
